Core/src/Core/Filters/Algorithms/CalculateArrayHistogram.cpp

```diff
@@ -0,0 +1,315 @@
+#include "CalculateArrayHistogram.hpp"
+
+#include "Core/Filters/CalculateArrayHistogramFilter.hpp"
+
+#include "complex/Common/Range.hpp"
+#include "complex/DataStructure/DataArray.hpp"
+#include "complex/DataStructure/DataGroup.hpp"
+#include "complex/Utilities/ParallelDataAlgorithm.hpp"
+
+#include <chrono>
+
+using namespace complex;
+
+namespace
+{
+template <typename T>
+class GenerateHistogramFromData
+{
+public:
+  GenerateHistogramFromData(CalculateArrayHistogram& filter, const int32 numBins, const DataArray<T>& inputArray, Float64Array& histogram, std::atomic<usize>& overflow, float64 minRange = 0.0,
+                            float64 maxRange = 0.0)
+  : m_Filter(filter)
+  , m_NumBins(numBins)
+  , m_Min(minRange)
+  , m_Max(maxRange)
+  , m_InputArray(inputArray)
+  , m_Histogram(histogram)
+  , m_Overflow(overflow)
+  {
+  }
+  ~GenerateHistogramFromData() = default;
+
+  void generate(const usize start, const usize end) const
+  {
+    // tuple visulaization: Histogram = {(bin maximum, count), (bin maximum, count), ... }
+
```

Comment:
This entire section can be handled by the [STL function min_max](https://en.cppreference.com/w/cpp/algorithm/minmax). complex DataArray has the proper iterators for this.